elastic/kibana
# [Index Management] Add data retention to component template UI

Closes: https://github.com/elastic/kibana/issues/165140

## Summary
This PR adds support for configuring data retention in component template UI. 

##### How to test
1. Start elasticsearch with `yarn es snapshot --license=trial -E path.repo=/tmp/snapshots` and kibana with `yarn start`
2. Navigate to index management, click on component template tab  and create a new one
3. Verify that:
  * Data retention can be configured and has a correct validation
  * Review template step should show the lifecycle values correctly in the preview and request tabs
  * Upon creating the template the details flyout should show the correct lifecycle value
  * Editing a template should initialize the values correctly


<details>
<summary>Screenshots</summary>

<img width="1235" alt="Screenshot 2023-11-08 at 10 54 20" src="https://github.com/elastic/kibana/assets/1191206/1e7f3bab-c903-480b-9da6-fd81b002eb62">
<img width="1237" alt="Screenshot 2023-11-08 at 10 54 33" src="https://github.com/elastic/kibana/assets/1191206/3681096f-53b5-4a61-9ae3-3bc16698f40e">
<img width="1242" alt="Screenshot 2023-11-08 at 10 54 43" src="https://github.com/elastic/kibana/assets/1191206/7edb846f-7334-4556-8e0a-608aa0adc6de">
<img width="1217" alt="Screenshot 2023-11-08 at 10 54 51" src="https://github.com/elastic/kibana/assets/1191206/da58890a-4e33-4085-87d2-90b8788286e8">
<img width="860" alt="Screenshot 2023-11-08 at 10 55 00" src="https://github.com/elastic/kibana/assets/1191206/aa1aba37-261f-46e8-b110-e015923ed7d3">



</details>
